gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/AuthConfig.groovy

```diff
@@ -16,35 +16,53 @@
 
 package com.netflix.spinnaker.gate.security
 
+import com.netflix.spinnaker.gate.security.anonymous.AnonymousConfig
+import groovy.util.logging.Slf4j
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.beans.factory.annotation.Qualifier
-import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression
 import org.springframework.boot.autoconfigure.security.SecurityAutoConfiguration
 import org.springframework.boot.context.embedded.FilterRegistrationBean
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 import org.springframework.context.annotation.Import
+import org.springframework.http.HttpMethod
+import org.springframework.security.authentication.AuthenticationManager
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder
 import org.springframework.security.config.annotation.web.builders.HttpSecurity
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+import org.springframework.security.core.userdetails.UserDetailsService
 import org.springframework.security.web.context.AbstractSecurityWebApplicationInitializer
 
 import javax.servlet.Filter
 
-@ConditionalOnExpression('${saml.enabled:false} || ${x509.enabled:false} || ${oauth2.enabled:false}')
```

Comment:
This turns this config on at all times. If none of the authentication mechanisms are enabled, this would render Deck useless. Is this the behavior we want?